File: src/dataset/python/space-age_exercism-ThomasZumsteg/openAI/ChatGPT4_space_age_v3.py
class SpaceAge:
    _earth_year = 31557600  # seconds
    _ratios = {
        "on_earth": 1.0,
        "on_mercury": 0.2408467,
        "on_venus": 0.61519726,
        "on_mars": 1.8808158,
        "on_jupiter": 11.862615,
        "on_saturn": 29.447498,
        "on_uranus": 84.016846,
        "on_neptune": 164.79132,
    }

    def __init__(self, seconds):
        self.seconds = seconds
        for planet, ratio in self._ratios.items():
            setattr(self, planet, self._calculate_age(ratio))

    def _calculate_age(self, ratio_to_earth):
        return lambda: round(self.seconds / (self._earth_year * ratio_to_earth), 2)

File: src/dataset/Java/space-age_exercism-ThomasZumsteg/openAI/ChatGPT4_SpaceAge_v3.java
public class SpaceAge {
    private final double seconds;
    private static final double EARTH_YEAR_IN_SECONDS = 31557600.0;
    private static final double MERCURY_ORBITAL_RATIO = 0.2408467;
    private static final double VENUS_ORBITAL_RATIO = 0.61519726;
    private static final double MARS_ORBITAL_RATIO = 1.8808158;
    private static final double JUPITER_ORBITAL_RATIO = 11.862615;
    private static final double SATURN_ORBITAL_RATIO = 29.447498;
    private static final double URANUS_ORBITAL_RATIO = 84.016846;
    private static final double NEPTUNE_ORBITAL_RATIO = 164.79132;

    public SpaceAge(Long seconds) {
        this.seconds = seconds;
    }

    public SpaceAge(Integer seconds) {
        this.seconds = seconds.doubleValue();
    }

    public Long getSeconds() {
        return (long) this.seconds;
    }

    public double onEarth() {
        return seconds / EARTH_YEAR_IN_SECONDS;
    }

    public double onMercury() {
        return seconds / (EARTH_YEAR_IN_SECONDS * MERCURY_ORBITAL_RATIO);
    }

    public double onVenus() {
        return seconds / (EARTH_YEAR_IN_SECONDS * VENUS_ORBITAL_RATIO);
    }

    public double onMars() {
        return seconds / (EARTH_YEAR_IN_SECONDS * MARS_ORBITAL_RATIO);
    }

    public double onJupiter() {
        return seconds / (EARTH_YEAR_IN_SECONDS * JUPITER_ORBITAL_RATIO);
    }

    public double onSaturn() {
        return seconds / (EARTH_YEAR_IN_SECONDS * SATURN_ORBITAL_RATIO);
    }

    public double onUranus() {
        return seconds / (EARTH_YEAR_IN_SECONDS * URANUS_ORBITAL_RATIO);
    }

    public double onNeptune() {
        return seconds / (EARTH_YEAR_IN_SECONDS * NEPTUNE_ORBITAL_RATIO);
    }
}